### ISSUE TITLE
United Kingdom

### ISSUE THREAD

**Issue by RobJN** [ideditor/country-coder#35]
The UK boundary is showing as including the British Overseas Territories and Crown dependencies. I believe this should be excluded. The have their own ISO codes.

https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%22GB%22%5D%7D

I think the region that it should show is:
https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%22GB-ENG%22%2C%20%22GB-NIR%22%2C%20%22GB-SCT%22%2C%20%22GB-WLS%22%5D%7D

Details on the British Overseas Territories and Crown dependencies can be found here:
https://en.wikipedia.org/wiki/United_Kingdom#Dependencies

**Comment by RobJN** [comment]
Related: https://github.com/osmlab/osm-community-index/pull/411

**Comment by quincylvania** [comment]
@RobJN Yep, this is a known issue I mentioned in https://github.com/ideditor/country-coder/issues/29#issuecomment-703336043. The problem with reversing the meaning is that there still wouldn't be an easy way to query all the places with UK sovereignty, unless we can find a Wikidata item that corresponds to that feature.

**Comment by RobJN** [comment]
Can we simply create such a wiki data entry? If so, I know just the person who could help.

**Comment by quincylvania** [comment]
> Can we simply create such a wiki data entry? If so, I know just the person who could help.

Sounds okay, if the Wikidata community is okay with it.

**Comment by pigsonthewing** [comment]
Here's your item; please check it includes what you expected:

   https://www.wikidata.org/wiki/Q105435336

Is there an equivalent OSM relation?

But what about https://www.openstreetmap.org/node/7685033716 ? ;-)

**Comment by pigsonthewing** [comment]
P.S. Note that crown dependencies are not, strictly speaking "UK sovereign territories".

**Comment by RobJN** [comment]
Hi @quincylvania. So the wikidata item you asked for now exists. How should we proceed? Are you going to make use of this new wikidata item for all the places with UK sovereignty and set the ISO code "GB" to what most would consider the UK to be? Reason I ask is that the community index entries for United Kingdom are all covering too big an area. I can either set them all to the right area like I did for the OSM UK local chapter (although without the Isle of Man or Channel Islands) but if you are making a change anyway I can leave them as is.

**Comment by bhousel** [comment]
Hey all, I'd like to close out this issue, but I really don't understand it.
From what I can tell, plenty of codes already exist to represent the various UK subregions.
https://ideditor.codes/ lists all the supported codes.

Region | locationSet | view it
--- | --- | ---
United Kingdom (includes everything) | `{ include: ['uk'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27uk%27%5D%20%7D%0A)
Great Britain (just the main island) | `{ include: ['Q23666'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27Q23666%27%5D%20%7D%0A)
Great Britain + Northern Ireland | `{ include: ['Q23666', 'GB-NIR'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27Q23666%27%2C%20%27GB-NIR%27%5D%20%7D%0A)
British Overseas Territories | `{ include: ['BOTS'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27BOTS%27%5D%20%7D%0A)
UK - BOTS<br/>(incl. Isle of Man and Channel Islands) | `{ include: ['UK'], exclude: ['BOTS'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27UK%27%5D%2C%20exclude%3A%20%5B%27BOTS%27%5D%20%7D%0A)

**Comment by RobJN** [comment]
Hi @bhousel 

>Hey all, I'd like to close out this issue, but I really don't understand it.

The issue is that a lot of UK entries in the Community and Imagery Indexs have used the locationSet `{ include: ['uk'] }` and all of these are wrong as this locationSet includes everything. Most UK citizens (and I expect most BOTS, Isle of Man and Channel Island citizens) would consider the UK to be what you have as locationSet `{ include: ['Q23666', 'GB-NIR'] }`. As noted in my first post, this also seems to be the case for ISO standards as the BOTS, Isle of Man and Channel Islands all have their own ISO codes.

From here we have two options:

1. Update locationSet `{ include: ['uk'] }` so that it covers the area that excludes he BOTS, Isle of Man and Channel Islands.
2. I go through and change all the Community Index records, plus all the Imagery Index records that currently use ['uk'] to instead be ['Q23666', 'GB-NIR']

Of these, option 1 was my preference as it is what makes most sense to the citizens of these territories and is in line with the ISO standards.

Quincy then replied:

> The problem with reversing the meaning [of 'uk'] is that there still wouldn't be an easy way to query all the places with UK sovereignty, unless we can find a Wikidata item that corresponds to that feature.

To satisfy Quincy's request, Andy (pigsonthewing) has created such a Wikidata entry that can be used for the full set (UK + BOTS + Isle of Man + Channel Islands).

My hope is that we can therefore proceed with option 1 above. If that is not reasonable then I will start the process of updating all the incorrect UK entries in the Community and Imagery indexes.

**Comment by bhousel** [comment]
It looks like we already support another one that I missed:

Region | locationSet | view it
--- | --- | ---
Countries of the United Kingdom | `{ include: ['Q3336843'] }` | [view it](https://ideditor.github.io/location-conflation/?locationSet=%7B%20include%3A%20%5B%27Q3336843%27%5D%20%7D%0A)

Which is the same thing as ` include: ['Q23666', 'GB-NIR']`

I'll try to:
- have `include: ['uk']` (and related codes) resolve to that geometry and qid instead, and 
- have `Q105435336` represent what `include: ['uk']` does currently (United Kingdom sovereignty)

**Comment by bhousel** [comment]
I tried this today, and the commit is saved at https://github.com/ideditor/country-coder/commit/706e8725e3f6772d67d72ce95706f277a2934ead, but it breaks a lot of the tests for this library and we'd need to redo some of the documentation too.  I am pretty sure I got a lot of it wrong, and I'm not really comfortable merging this into main.

@RobJN I think the faster solution for now is your "option 2" - to just search-and-replace the locationSets in those other projects to say `{ include: ['Q3336843'] }`.

**Comment by RobJN** [comment]
Ok @bhousel. Thanks for trying. I will get on to option 2 but with a couple weeks ahead it will be late July / early August now.

**Comment by bhousel** [comment]
I just took care of this in the osm-community-index project, it was only a few files..
